internal/attestation/crafter/materials/csaf.go

```diff
@@ -0,0 +1,125 @@
+func NewCSAFVEXCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_CSAF_VEX {
+		return nil, fmt.Errorf("material type is not CSAF_VEX format")
+	}
+
+	return baseCSAFCrafter(materialSchema, backend, CategoryVEX, l)
+}
+
+// NewCSAFInformationalAdvisoryCrafter creates a new CSAF Informational Advisory crafter
+func NewCSAFInformationalAdvisoryCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_CSAF_INFORMATIONAL_ADVISORY {
+		return nil, fmt.Errorf("material type is not CSAF_INFORMATIONAL_ADVISORY format")
+	}
+
+	return baseCSAFCrafter(materialSchema, backend, CategoryInformationalAdvisory, l)
+}
+
+// NewCSAFSecurityIncidentResponseCrafter creates a new CSAF Security Incident Response crafter
+func NewCSAFSecurityIncidentResponseCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_CSAF_SECURITY_INCIDENT_RESPONSE {
+		return nil, fmt.Errorf("material type is not CSAF_SECURITY_INCIDENT_RESPONSE format")
+	}
+
+	return baseCSAFCrafter(materialSchema, backend, CategorySecurityIncidentResponse, l)
+}
+
+// NewCSAFSecurityAdvisoryCrafter creates a new CSAF Security Advisory crafter
+func NewCSAFSecurityAdvisoryCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, l *zerolog.Logger) (*CSAFCrafter, error) {
+	if materialSchema.Type != schemaapi.CraftingSchema_Material_CSAF_SECURITY_ADVISORY {
+		return nil, fmt.Errorf("material type is not CSAF_SECURITY_ADVISORY format")
+	}
+
+	return baseCSAFCrafter(materialSchema, backend, CategorySecurityAdvisory, l)
+}
+
+// NewCSAFCrafter creates a new CSAF crafter
+func baseCSAFCrafter(materialSchema *schemaapi.CraftingSchema_Material, backend *casclient.CASBackend, category string, l *zerolog.Logger) (*CSAFCrafter, error) {
+	return &CSAFCrafter{
+		backend:       backend,
+		crafterCommon: &crafterCommon{logger: l, input: materialSchema},
+		category:      category,
+	}, nil
+}
+
+func (i *CSAFCrafter) Craft(ctx context.Context, filepath string) (*api.Attestation_Material, error) {
+	i.logger.Debug().Str("path", filepath).Msg("decoding CSAF file")
+	f, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, fmt.Errorf("can't open the file: %w - %w", err, ErrInvalidMaterialType)
+	}
+
+	var v interface{}
+	if err := json.Unmarshal(f, &v); err != nil {
+		i.logger.Debug().Err(err).Msg("error decoding file")
+		return nil, fmt.Errorf("invalid CSAF file: %w", ErrInvalidMaterialType)
+	}
+
+	// Validate the CSAF file against the specified category.
+	document, docExists := v.(map[string]interface{})["document"]
```

Comment:
it would have been more clear if we use a custom struct with only the document and category objects.

Reply:
hmm but the custom struct won't have the remaining fields to be passed down to the validator no? Maybe I'm missing something :/

Reply:
it will not, that's why I mentioned doing this validation after the other one. You might require another marshall process though.

Anyways, not a big deal, feel free to merge it :)